src/cpp/src/visual_language/minicpm/classes.cpp

```diff
@@ -749,7 +732,62 @@ ov::Tensor InputsEmbedderMiniCPM::resample(const ov::Tensor& encoded_image, cons
     resampler.set_tensor("key_padding_mask", key_padding_mask);  // [N, H*W]
     resampler.start_async();
     resampler.wait();
-    return resampler.get_output_tensor();  // [N, query_num, new_hidden_size]
+    auto resampler_out = resampler.get_output_tensor();
+    // resampler_out is bound to infer request and the data may become corrupted after next resampler inference 
+    // so we need to return a copy to make sure data does not get corrupted 
+    ov::Tensor res(resampler_out.get_element_type(), resampler_out.get_shape());
+    std::memcpy(res.data(), resampler_out.data(), resampler_out.get_byte_size());
+    return res;  // [N, query_num, new_hidden_size]
```

Comment:
don't we need to perform std::move? Otherwise, it makes additional copy.

Reply:
We need a copy here, otherwise next infer request corrupts the tensor data. I left the comment about this above.

Reply:
Return values are "moved" by default. It's called (named) return value optimization. `std::move()` only hurts that optimization.